cachelib/allocator/CacheAllocator-inl.h

```diff
@@ -1526,17 +1548,33 @@ CacheAllocator<CacheTrait>::createPutToken(Item& item) {
 }
 
 template <typename CacheTrait>
-void CacheAllocator<CacheTrait>::unlinkItemForEviction(Item& it) {
+bool CacheAllocator<CacheTrait>::unlinkItemForEviction(Item& it) {
   XDCHECK(it.isMarkedForEviction());
   XDCHECK(it.getRefCount() == 0);
-
-  accessContainer_->remove(it);
+  bool removedFromNvm = false;
+  if (!accessContainer_->remove(it)) {
+      //item is unaccessible in in hash table meaning that
+      //insertOrReplace was called - we need to delete this
+      //from NVM
+      XDCHECK(!it.isAccessible());
+      HashedKey hk{it.getKey()};
+      auto lock = nvmCache_ ? nvmCache_->getItemDestructorLock(hk)
+                            : std::unique_lock<std::mutex>();
+      if (it.isNvmClean() && !it.isNvmEvicted()) {
+        // item is to be replaced and the destructor will be executed
+        // upon memory released, mark it in nvm to avoid destructor
+        // executed from nvm
+        nvmCache_->markNvmItemRemovedLocked(hk);
```

Comment:
Do we need to call this here? Isn't the call in `insertOrReplace` enough?

Reply:
Yes as we discussed - I fixed in latest commit - https://github.com/intel/CacheLib/pull/56/commits/5eb049e97fa24e8bced785e392ebc2437d0f4b25